luminoth/tools/dataset/readers/object_detection/imagenet.py

```diff
@@ -0,0 +1,209 @@
+
+        # Load wnids from file.
+        wnids_path = os.path.join(
+            os.path.dirname(os.path.realpath(__file__)),
+            WNIDS_FILE
+        )
+        with tf.gfile.GFile(wnids_path) as wnidsjson:
+            self._wnids = json.load(wnidsjson)
+
+    @property
+    def total(self):
+        if self._total is None:
+            total_records = 0
+            for line in self._get_record_names():
+                total_records += 1
+
+            # Define smaller number of records when limiting examples.
+            if self._only_filename is not None:
+                self._total = 1
+            elif self._limit_examples is not None and self._limit_examples > 0:
+                self._total = min(self._limit_examples, total_records)
+            else:
+                self._total = total_records
+
+        return self._total
+
+    @property
+    def classes(self):
+        if self._classes is None:
+            self._classes = sorted(list(self._wnids.values()))
+
+            # Choose random classes when limiting them
+            if self._limit_classes is not None and self._limit_classes > 0:
+                total_classes = min(len(self._classes), self._limit_classes)
+                self._classes = sorted(
+                    random.sample(self._classes, total_classes)
+                )
+
+        return self._classes
+
+    def iterate(self):
+        for image_id in self._get_record_names():
+            # Ignore image_id is using only_filename filter and it's different.
+            if self._only_filename and image_id != self._only_filename:
+                continue
+
+            if self.yielded_records == self.total:
+                # Finish iteration based on predefined total.
+                return
+
+            try:
+                annotation_path = self._get_image_annotation(image_id)
+                image_path = self._get_image_path(image_id)
+                # Read both the image and the annotation into memory.
+                annotation = read_xml(annotation_path)
+                image = read_image(image_path)
+            except tf.errors.NotFoundError:
+                tf.logging.debug(
+                    'Error reading image or annotation for "{}".'.format(
+                        image_id))
+                self.errors += 1
+                continue
+
+            objects = annotation.get('object')
+            if objects is None:
+                # If there's no bounding boxes, we don't want it.
+                continue
+
+            # TODO: consider alternatives to using Pillow here.
+            image_pil = Image.open(image_path)
+            width = image_pil.width
+            height = image_pil.height
+            image_pil.close()
+
+            gt_boxes = []
+            for b in annotation['object']:
+                try:
+                    label_id = self.classes.index(self._wnids[b['name']])
+                except ValueError:
+                    continue
```

Comment:
Not sure, but what is happening here? Should we warn or something? Is this expected? Add comment.